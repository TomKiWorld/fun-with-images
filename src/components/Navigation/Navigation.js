import React from 'react';
import Menu from '../Menu/Menu';

/**
 * Site navigation
 * 
 * Required props:
 * - onRouteChange => Function to change the state of route
 * - isSignedIn => State to show relative links
 */
const Navigation = ({ onRouteChange, isSignedIn }) => {
  if (isSignedIn) {
    return (
      <nav className='main-nav end'>
        <Menu
          onRouteChange={onRouteChange} />
      </nav>
    )
  } else {
    return(
      <nav className='main-nav end'>
        <p 
          onClick={() => onRouteChange('signin')}
          className='link dim black underline pa3 pointer' >Sign in</p>
        <p 
          onClick={() => onRouteChange('register')}
          className='link dim black underline pa3 pointer' >Register</p>
      </nav>
    )
  }
}

export default Navigation;
