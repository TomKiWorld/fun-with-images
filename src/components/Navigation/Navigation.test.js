import { shallow } from 'enzyme';
import React from 'react';
import Navigation from './Navigation';

describe('Navigation Component', () => {
  const mockState = {
    route: 'signin',
    isSignedIn: false
  };

  const mockRouteChange = (route) => {
    mockState.route = route;
  }

  const wrapper = shallow(
    <Navigation 
      route={mockState.route} 
      isSignedIn={mockState.isSignedIn}
      onRouteChange={(value) => mockRouteChange(value)}
    />
  );

  it('Expect to render Navigation Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Expect to change rout to register', () => {
    const registerBtn = wrapper.find('.link').at(1);
    registerBtn.simulate('click', { value: 'register' } );
    expect(mockState.route).toBe('register');
  });

  it('Expect to change rout to sigin', () => {
    const signinBtn = wrapper.find('.link').at(0);
    signinBtn.simulate('click', { value: 'signin' } );
    expect(mockState.route).toBe('signin');
  });

  it('Expect to render Navigation Component when signed in', () => {
    const wrapper = shallow(
      <Navigation 
        route={mockState.route} 
        isSignedIn={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
