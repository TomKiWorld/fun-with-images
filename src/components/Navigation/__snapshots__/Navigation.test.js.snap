// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component Expect to render Navigation Component 1`] = `ShallowWrapper {}`;

exports[`Navigation Component Expect to render Navigation Component when signed in 1`] = `ShallowWrapper {}`;
